### ISSUE TITLE
Thread exception while jobs -K

### ISSUE THREAD

**Issue by wchen-r7** [rapid7/metasploit-framework#5675]
There seems to be a problem with how jobs are closed. All you have to do is start an exploit, and then kill the job, and you will see this problem:

1. Start msfconsole
2. use exploit/windows/browser/ms14_064_ole_code_execution
3. jobs -K
4. Check framework.log

Backtraces:

```
[07/07/2015 00:41:11] [e(0)] core: thread exception: JobID(0)-Exploit: windows/browser/ms14_064_ole_code_execution  critical=false  error: IOError closed stream
  source:
    /Users/sinn3r/rapid7/msf/lib/metasploit/framework/thread_factory_provider.rb:24:in `spawn'
    /Users/sinn3r/rapid7/msf/lib/rex/thread_factory.rb:22:in `spawn'
    /Users/sinn3r/rapid7/msf/lib/rex/job_container.rb:32:in `start'
    /Users/sinn3r/rapid7/msf/lib/rex/job_container.rb:156:in `start_bg_job'
    /Users/sinn3r/rapid7/msf/lib/msf/core/exploit_driver.rb:149:in `run'
    /Users/sinn3r/rapid7/msf/lib/msf/base/simple/exploit.rb:136:in `exploit_simple'
    /Users/sinn3r/rapid7/msf/lib/msf/base/simple/exploit.rb:161:in `exploit_simple'
    /Users/sinn3r/rapid7/msf/lib/msf/ui/console/command_dispatcher/exploit.rb:111:in `cmd_exploit'
    /Users/sinn3r/rapid7/msf/lib/rex/ui/text/dispatcher_shell.rb:427:in `run_command'
    /Users/sinn3r/rapid7/msf/lib/rex/ui/text/dispatcher_shell.rb:389:in `block in run_single'
    /Users/sinn3r/rapid7/msf/lib/rex/ui/text/dispatcher_shell.rb:383:in `each'
    /Users/sinn3r/rapid7/msf/lib/rex/ui/text/dispatcher_shell.rb:383:in `run_single'
    /Users/sinn3r/rapid7/msf/lib/rex/ui/text/shell.rb:203:in `run'
    /Users/sinn3r/rapid7/msf/lib/metasploit/framework/command/console.rb:48:in `start'
    /Users/sinn3r/rapid7/msf/lib/metasploit/framework/command/base.rb:82:in `start'
    ./msfconsole:48:in `<main>'
[07/07/2015 00:41:11] [e(0)] core: Call Stack
/Users/sinn3r/rapid7/msf/lib/msf/core/handler/reverse_tcp.rb:258:in `close'
/Users/sinn3r/rapid7/msf/lib/msf/core/handler/reverse_tcp.rb:258:in `stop_handler'
/Users/sinn3r/rapid7/msf/lib/msf/core/exploit_driver.rb:321:in `job_cleanup_proc'
/Users/sinn3r/rapid7/msf/lib/msf/core/exploit_driver.rb:153:in `block in run'
/Users/sinn3r/rapid7/msf/lib/rex/job_container.rb:39:in `call'
/Users/sinn3r/rapid7/msf/lib/rex/job_container.rb:39:in `ensure in block in start'
/Users/sinn3r/rapid7/msf/lib/rex/job_container.rb:40:in `block in start'
/Users/sinn3r/rapid7/msf/lib/rex/thread_factory.rb:22:in `call'
/Users/sinn3r/rapid7/msf/lib/rex/thread_factory.rb:22:in `block in spawn'
/Users/sinn3r/rapid7/msf/lib/msf/core/thread_manager.rb:100:in `call'
/Users/sinn3r/rapid7/msf/lib/msf/core/thread_manager.rb:100:in `block in spawn'
```

**Comment by bcook-r7** [comment]
Let's get a handle on the frequency with which this would hit before we prioritize this.

**Comment by wchen-r7** [comment]
Resolved in #5650

Please use module: auxiliary/server/browser_autopwn2
